src/method.c

```diff
@@ -814,7 +821,9 @@ JL_DLLEXPORT jl_method_t* jl_method_def(jl_svec_t *argdata,
         argtype = jl_new_struct(jl_unionall_type, tv, argtype);
     }
 
-    jl_methtable_t *mt = jl_method_table_for(argtype);
+    jl_methtable_t *external_mt = mt;
+    if (!mt)
+        mt = jl_method_table_for(argtype);
```

Comment:
Can't we make `ccall(:jl_method_def, Cvoid, (Any,Any,Any,Any), sig, false, body, module)` work ?
This C function seems to be used by [multiple packages](https://juliahub.com/ui/Search?q=%3Ajl_method_def&type=code), but it seems that we need some workaround like:
```julia
        atypes = sig[1]::SimpleVector
        mt = ccall(:jl_method_table_for, Any, (Any,), Tuple{atypes...})
        ccall(:jl_method_def, Cvoid, (Any,Any,Any,Any), sig, mt, body, moduleof(frame))
```
(xref: https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/489)

Reply:
nvm, @simeonschaub showed me how it should be called:
```julia
ccall(:jl_method_def, Cvoid, (Any,Ptr{Cvoid},Any,Any), sig, C_NULL, body, moduleof(frame))
```
https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/488#issuecomment-858568221